File: ai/ai-best-practices/openai-missing-safety-identifier/openai-missing-safety-identifier.js
const OpenAI = require("openai");
const client = new OpenAI();

async function test() {
    // ruleid: openai-missing-safety-identifier-javascript
    const response = await client.responses.create({
        model: "gpt-4.1",
        input: "Hello, how are you?"
    });

    // ok: openai-missing-safety-identifier-javascript
    const response2 = await client.responses.create({
        model: "gpt-4.1",
        input: "Hello, how are you?",
        safety_identifier: "user_abc123"
    });
}

File: ai/ai-best-practices/openai-missing-safety-identifier/openai-missing-safety-identifier.py
from openai import OpenAI

client = OpenAI()

# ruleid: openai-missing-safety-identifier-python
response = client.responses.create(
    model="gpt-4.1",
    input="Hello, how are you?"
)

# ok: openai-missing-safety-identifier-python
response = client.responses.create(
    model="gpt-4.1",
    input="Hello, how are you?",
    safety_identifier="user_abc123"
)
